File: 연습/재귀/부분수열의합14225/부분수열의합.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int solve(vector<int>& a, int index, int target){
  if(a[index]>target) return target;
  if(index==a.size()) return target;
  return solve(a,index+1,target+a[index]);
}

int main(){
  int n; 
  cin>>n;
  vector<int> a(n);
  for(int i=0; i<n; i++){
    cin>>a[i];
  }
  sort(a.begin(),a.end());
  cout<<solve(a,0,1)<<'\n';
}

File: 연습/재귀/부분수열의합14225/부분수열의합.py
def solve(a,index,target):
  if index==len(a) : return target
  if a[index] > target: return target
  return solve(a,index+1,target+a[index])
n=int(input())
a=list(map(int,input().split()))
a.sort()
print(solve(a,0,1))